app/src/main/java/com/hover/stax/bounties/BountyListFragment.java

```diff
@@ -48,17 +57,28 @@ private void startObservers() {
 		bountyViewModel.getActions().observe(getViewLifecycleOwner(), actions -> Log.v(TAG, "actions update: " + actions.size()));
 		bountyViewModel.getTransactions().observe(getViewLifecycleOwner(), transactions -> Log.v(TAG, "transactions update: " + transactions.size()));
 		bountyViewModel.getBounties().observe(getViewLifecycleOwner(), bounties -> {
-			if (bounties != null && bounties.size() > 0 && bountyViewModel.getChannels().getValue() != null && bountyViewModel.getChannels().getValue().size() > 0)
-				createList(bountyViewModel.getChannels().getValue(), bountyViewModel.getBounties().getValue());
+			List<Channel> channels = bountyViewModel.getChannels().getValue();
+			if (bounties != null && bounties.size() > 0 && channels != null && bountyViewModel.getChannels().getValue().size() > 0){
+				createList(channels, bountyViewModel.getBounties().getValue(), true);
+			}
 		});
 
 		bountyViewModel.getChannels().observe(getViewLifecycleOwner(), channels -> {
-			if (channels != null && channels.size() > 0 && bountyViewModel.getBounties().getValue() != null && bountyViewModel.getBounties().getValue().size() > 0)
-				createList(channels, bountyViewModel.getBounties().getValue());
+			if (channels != null && channels.size() > 0 && bountyViewModel.getBounties().getValue() != null && bountyViewModel.getBounties().getValue().size() > 0) {
+				createList(channels, bountyViewModel.getBounties().getValue(), true);
+			}
+		});
+	}
+	private void filterBounties(String countryCode) {
+		bountyViewModel.filterChannels(countryCode).observe(getViewLifecycleOwner(), channels -> {
+			if (channels != null && channels.size() > 0 && bountyViewModel.getBounties().getValue() != null && bountyViewModel.getBounties().getValue().size() > 0) {
```

Comment:
Since we are repeating this same null check everywhere, move it into the `createList` function. Also rename createList, bad naming on my part.